core/controllers/learner_group.py

```diff
@@ -314,6 +314,74 @@ def get(self, learner_group_id):
         self.render_json(all_learners_progress)
 
 
+class LearnerGroupLearnerSpecificProgressHandler(base.BaseHandler):
+    """Handles operations related to fetching learner specific progress."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    URL_PATH_ARGS_SCHEMAS = {
+        'learner_group_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.LEARNER_GROUP_ID_REGEX
+                }]
+            },
+            'default_value': None
+        }
+    }
+
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {}
+    }
+
+    @acl_decorators.can_access_learner_groups
+    def get(self, learner_group_id):
+        """Handles GET requests for user progress through learner
+        group syllabus.
+        """
+
+        learner_user_id = self.user_id
+
+        learner_group = learner_group_fetchers.get_learner_group_by_id(
+            learner_group_id)
```

Comment:
if `learner_group_id` is None then why we are passing this to `get_learner_group_by_id` function ?, because `get_learner_group_by_id` function can only accept string values ( this is what written in docstring and typing )

Reply:
removed the default value none so the check is not needed